Fix bug in upgrader generation in mobile

tugsbayasgalan opened the pull request:
Summary: Use more robust way of extracting upgrader min and max versions

Test Plan: omgitsgreen

Differential Revision: D33690113

cccclai commented on torch/csrc/jit/mobile/upgrader_mobile.cpp:
```diff
@@ -1,257 +1,302 @@
+
 #include <torch/csrc/jit/mobile/upgrader_mobile.h>
 
-namespace c10 {
-TypePtr parseType(const std::string& pythonStr);
-} // namespace c10
```
Looks like these lines were missing in the codegen, could you help with adding them?

cccclai replied:
I meant adding them as part of `gen_mobile_upgraders.py`

cccclai commented on torch/csrc/jit/mobile/upgrader_mobile.cpp:
```diff
@@ -1,257 +1,302 @@
-namespace c10 {
-TypePtr parseType(const std::string& pythonStr);
-} // namespace c10
+#include <ATen/core/ivalue.h>
+#include <caffe2/serialize/versions.h>
+#include <torch/csrc/jit/mobile/type_parser.h>
```
These headers may not be needed, if the previous version work.

cccclai commented on torch/_C/__init__.pyi.in:
```diff
@@ -462,8 +462,14 @@ class _UpgraderEntry:
+
 def _get_operator_version_map() -> Dict[str, List[_UpgraderEntry]]: ...
 
+def _get_upgrader_ranges(name: str) -> List[_UpgraderRange]: ...
```
Thanks for creating an api for it! Very neat

cccclai commented on tools/codegen/operator_versions/gen_mobile_upgraders.py:
```diff
@@ -217,48 +217,27 @@ def construct_register_size(register_size_from_yaml: int) -> str:
-            upgrader_info = list(filter(lambda token: token != "", upgrader_entry.upgrader_name.split('_')))
-            upgrader_min_version = upgrader_info[2]
-            upgrader_max_version = upgrader_info[3]
+        upgrader_ranges = torch._C._get_upgrader_ranges(op_name)
```
Neat. Thank!

cccclai reviewed (approved):
LGTM. Please also check CI. Thanks!

tugsbayasgalan committed:
Fix bug in upgrader generation in mobile (#71578)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/71578

Use more robust way of extracting upgrader min and max versions

Test Plan: omgitsgreen

Reviewed By: cccclai

Differential Revision: D33690113

fbshipit-source-id: 5c1f7b90bcd5b57f9984a99ff8f7a7aad1a3c7d8
